ldms/src/core/ldms_xprt.c

```diff
@@ -340,7 +340,7 @@ static void send_dir_update(struct ldms_xprt *x,
 	hdr_len = sizeof(struct ldms_reply_hdr)
 		+ sizeof(struct ldms_dir_reply);
 	buf_len = ldms_xprt_msg_max(x);
-	reply = malloc(buf_len);
+	reply = calloc(1, buf_len);
```

Comment:
@ballan, This is a bzero of a megabyte on the socket and uGNI transports. The only bits that won't be initialized are the padding bytes in the header. Can you please just bzero the header (24B) and rerun valgrind to confirm this?

Reply:
@tom95858 I can try that, but the error message says the issue is '96 bytes inside a block of 420', which appears to be the middle of the json space. I will rerun the tests; it may be that subsequent changes to json payload have caused the issue to disappear. I need to verify my test is still running the same sampler set as well, if something is payload dependent.

Reply:
Did you confirm that this change fixed the valgrind complaint? Regardless we can't bzero 1MB of data on every message, you need to figure out what the actual problem is.

Reply:
If the problem is in the JSON payload, then this change will not fix it because the json payload is copied into the 1MB buffer calloc bzero'd. Also, the error would be '96B inside a block of 1MB, not 420B. What transport was this problem exhibited on?

Reply:
I am working on getting my hands on the data where the hole is found. I'm seeing this kind of message on both sock and rdma on stria. Started adding more logging yesterday, and will also see about getting valgrind+gdb to stop on detection and let me look around (though that's always messier with multithreaded code).